x-pack/plugins/alerting/server/application/rule/methods/update/update_rule.ts

```diff
@@ -114,8 +141,24 @@ async function updateWithOCC<Params extends RuleParams = never>(
     systemActions: genSystemActions,
   };
 
-  const { alertTypeId, consumer, enabled, schedule, name, apiKey, apiKeyCreatedByUser } =
-    originalRuleSavedObject.attributes;
+  const {
+    alertTypeId,
+    consumer,
+    enabled,
+    schedule,
+    name,
+    apiKey,
+    apiKeyCreatedByUser,
+    flapping: originalFlapping,
+  } = originalRuleSavedObject.attributes;
+
+  validateCanUpdateFlapping(isFlappingEnabled, originalFlapping, initialData.flapping);
+
+  if (!isFlappingEnabled && !isEqual(originalFlapping, initialData.flapping)) {
```

Comment:
nit: It seems this scenario is covered by `validateCanUpdateFlapping`, right? If not can we move it inside the function?

Reply:
good catch